roc_example_test.go

```diff
@@ -77,7 +79,8 @@ func ExampleROC_equallySpacedCutoffs() {
 	n := 9
 
 	stat.SortWeightedLabeled(y, classes, weights)
-	cutoffs := stat.EquallySpaced(y[0], y[len(y)-1], n)
+	cutoffs := make([]float64, n)
+	floats.Span(cutoffs, math.Nextafter(y[0], y[0]-1), y[len(y)-1])
```

Comment:
right, so if the cutoffs are specified by the user its all good -- and no intervals are really involved. The issue is when cutoffs is nil or empty -- in which case ROC defaults to calculating all possible cutoffs, and therein if n y-values are included, y_1, y_2, ..., y_n, then this creates n+1 intervals, where cutoffs in any one of these intervals will produce the same fpr and tpr, so a maximum of n+1 unique (fpr, tpr) pairs. (n-1) of which are of the half-open form [y_k, y_{k+1}) for k = 1, 2, ..., n-1, then there are the two other `special' intervals, (-Inf, y_1) and [y_n, Inf).